platform/security/handler/security-handler-api/src/main/java/org/codice/ddf/security/handler/api/BaseAuthenticationTokenFactory.java

```
@@ -0,0 +1,193 @@
+    } catch (WSSecurityException | IOException e) {
+      LOGGER.warn("Unable to read merlin properties file. Unable to validate certificates.", e);
+    }
+    Init.init();
+  }
+
+  /**
+   * Creates a {@link BaseAuthenticationToken} from a given username and password. Uses a {@link
+   * UsernameTokenType} internally to store the username and password.
+   *
+   * @param username - user's username
+   * @param password - user's password
+   * @return a BaseAuthenticationToken containing the given username and password
+   */
+  public BaseAuthenticationToken getTokenFromUsernamePassword(String username, String password) {
+    UsernameTokenType usernameTokenType = new UsernameTokenType();
+    AttributedString user = new AttributedString();
+    user.setValue(username);
+    usernameTokenType.setUsername(user);
+    String usernameToken = null;
+
+    // Add a password
+    PasswordString pass = new PasswordString();
+    pass.setValue(password);
+    pass.setType(WSConstants.PASSWORD_TEXT);
+    JAXBElement<PasswordString> passwordType =
+        new JAXBElement<>(QNameConstants.PASSWORD, PasswordString.class, pass);
+    usernameTokenType.getAny().add(passwordType);
+
+    // Marshall the received JAXB object into a DOM Element
+    List<String> ctxPath = new ArrayList<>(2);
+    ctxPath.add(ObjectFactory.class.getPackage().getName());
+    ctxPath.add(
+        org.apache.cxf.ws.security.sts.provider.model.wstrust14.ObjectFactory.class
+            .getPackage()
+            .getName());
+
+    ParserConfigurator configurator =
+        parser.configureParser(ctxPath, BaseAuthenticationToken.class.getClassLoader());
+    ByteArrayOutputStream os = new ByteArrayOutputStream();
+
+    JAXBElement<UsernameTokenType> tokenType =
+        new JAXBElement<>(
+            QNameConstants.USERNAME_TOKEN, UsernameTokenType.class, usernameTokenType);
+
+    try {
+      parser.marshal(configurator, tokenType, os);
+      usernameToken = os.toString("UTF-8");
```

❓ If this fails to parse do we want to still return the token? Or would it be better to return an exception and let the caller handle it?